src/pages/resultsView/mutation/ResultsViewMutationTable.tsx

```diff
@@ -62,8 +74,30 @@ export default class ResultsViewMutationTable extends MutationTable<
             MutationTableColumnType.DBSNP,
             MutationTableColumnType.SIGNAL,
         ],
+        columnVisibilityProps: {},
     };
 
+    componentDidMount() {
+        this.props.columnVisibilityProps!.customDropdown = (
```

Comment:
Is it possible to do this in the render cycle instead of `componentDidMount`? So, somewhere something like this: 

```JavaScript
columnVisibilityProps={
    ...this.props.columnVisibilityProps,
    customDropdown: ( ... )
}
```

Reply:
I'm unable to change `columnVisibilityProps` since props are readonly. I've moved this to the constructor instead of `componentDidMount`. I don't think I can/would want to put it in the render cycle since it only passes the dropdown component once and then the rendering is handled in that component itself. Please let me know if I'm misunderstanding your request.

Reply:
Defining and passing all properties of the Column Visibility component at a single place would be ideal but I guess it is tricky to do it that way with the current design. I think it is okay to modify the props object in the constructor for now.